### PR TITLE
add new endpoint HEAD get file metadata

### PR DESCRIPTION
**Pull request by bakhdaulet**

### CODE REVIEW

File: repository_files.go [hunk 1]
```diff
@@ -48,6 +49,9 @@ func (r File) String() string {
 	return Stringify(r)
 }
 
+
+
+
```
Comment (svanharmelen) [inline repository_files.go]: Can you remove these 3 blank lines?

File: repository_files.go [hunk 2]
```diff
@@ -82,6 +86,54 @@ func (s *RepositoryFilesService) GetFile(pid interface{}, fileName string, opt *
 	return f, resp, err
 }
 
+
+
+
+
```
Comment (svanharmelen) [inline repository_files.go]: Can you remove these 4 blank lines?

File: repository_files.go [hunk 3]
```diff
@@ -82,6 +86,54 @@ func (s *RepositoryFilesService) GetFile(pid interface{}, fileName string, opt *
...
+// GetFileOptions represents the available GetFile() options.
```
Comment (svanharmelen) [inline repository_files.go]: This should be `GetFileMetaOptions`

File: repository_files.go [hunk 4]
```diff
@@ -82,6 +86,54 @@ func (s *RepositoryFilesService) GetFile(pid interface{}, fileName string, opt *
...
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/repository_files.html#get-file-from-repository
+type GetFileMetaOptions struct {
+	Ref *string `url:"ref,omitempty" json:"ref,omitempty"`
+}
+
+// GetFile allows you to receive information about a file in repository like
```
Comment (svanharmelen) [inline repository_files.go]: This should be `GetFileMeta`, but maybe we should call them (this one and the options struct `GetFileMetaData`? Sounds better to me...

File: repository_files.go [hunk 5]
```diff
@@ -82,6 +86,54 @@ func (s *RepositoryFilesService) GetFile(pid interface{}, fileName string, opt *
...
+	}
+
+	f.BlobID = resp.Header.Get("X-Gitlab-Blob-Id")
+	f.CommitID = resp.Header.Get("X-Gitlab-Last-Commit-Id")
+	f.Encoding = resp.Header.Get("X-Gitlab-Encoding")
+	f.FileName = resp.Header.Get("X-Gitlab-File-Name")
+	f.FilePath = resp.Header.Get("X-Gitlab-File-Path")
+	f.Ref = resp.Header.Get("X-Gitlab-Ref")
+	if sizeString := resp.Header.Get("X-Gitlab-Size"); sizeString != "" {
+		f.Size,_ = strconv.Atoi(sizeString)
```
Comment (svanharmelen) [inline repository_files.go]: Could you use `gofmt` to format your code? And are you sure we can safely ignore the error here?

File: repository_files.go (hunk 1 above)
Comment (bakhdaulet) [reply repository_files.go]: removed

File: repository_files.go (hunk 2 above)
Comment (bakhdaulet) [reply repository_files.go]: removed

File: repository_files.go (hunk 3 above)
Comment (bakhdaulet) [reply repository_files.go]: changed

File: repository_files.go (hunk 4 above)
Comment (bakhdaulet) [reply repository_files.go]: changed

File: repository_files.go (hunk 5 above)
Comment (bakhdaulet) [reply repository_files.go]: added check error

**Comment by svanharmelen** [comment]
Thanks @bakhdaulet! LGTM!

**Pull request merged**